common/k8s/src/middleware/metadata.rs

```diff
@@ -104,11 +231,130 @@ impl K8sMetadata {
             ListParams::default()
         };
 
-        let watched = StreamBackoff::new(watcher(api, params), ExponentialBackoff::default());
+        let watcher = Box::pin(StreamBackoff::new(
+            watcher(api, params),
+            ExponentialBackoff::default(),
+        ));
+
+        let (deletion_ack_recv, deletion_state) = {
+            let state = self.state.lock().map_err(|_| Error::K8sMiddlewareState)?;
+            // Take the deletion ack stream, copy the deletion state stream
+            (self.deletion_ack_recv.clone(), state.deletion_state.clone())
+        };
+
+        let delete_stream = stream::unfold(
+            (deletion_state.clone(), deletion_ack_recv),
+            move |(deletion_state, deletion_ack_recv)| async move {
+                let next_event = loop {
+                    if let Ok(Some((deleted_container, deleted_container_pod))) = deletion_ack_recv
```

Comment:
Err possible?

Reply:
Same comment as above. An error condition is unlikely. I added so warning logs in case an error does occur.